### PR TITLE
ci: add action for python linting and formatting

### PR DESCRIPTION
**Pull request by nathan-weinberg**
Implemented this locally in #7 but missed this action

### CODE REVIEW

**Comment by nathan-weinberg** [comment]
Only CI failure here is due to #6

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# SPDX-License-Identifier: Apache-2.0
+click>=8.1.7,<9.0.0
+httpx>=0.25.0,<1.0.0
+jinja2
+openai>=1.13.3,<2.0.0
+rouge_score
+tqdm>=4.66.2,<5.0.0
```
Comment (russellb) [inline requirements.txt:7]: can you add `instructlab` here?  until we resolve #6

**Comment by bjhargrave** [comment]
We should also add the `mypy` step from https://github.com/instructlab/instructlab/pull/1235 once it is completed.

**Comment by tiran** [comment]
You also need to add a `py.typed` marker file to `src/instructlab/sdg`, so the parent package knows that the subpackage uses type annotations. https://peps.python.org/pep-0561/#packaging-type-information

File: requirements.txt (hunk 1 above)
Comment (nathan-weinberg) [reply requirements.txt:7]: Yup, added that and a note to remove when the issue is resolved

**Comment by nathan-weinberg** [comment]
> We should also add the `mypy` step from [instructlab/instructlab#1235](https://github.com/instructlab/instructlab/pull/1235) once it is completed.

As part of this PR or as a followup?

> You also need to add a `py.typed` marker file to `src/instructlab/sdg`, so the parent package knows that the subpackage uses type annotations. https://peps.python.org/pep-0561/#packaging-type-information

That path doesn't exist - do you mean `src/instructlab_sdg`? What should this file look like?

**Comment by nathan-weinberg** [comment]
@bjhargrave since the MyPy PR merged in the other repo I went ahead and added it to this PR

**Review by bjhargrave — APPROVED** [review]

**Review by russellb — APPROVED** [review]

**Pull request merged**